File: 0038-count-and-say/0038-count-and-say.java
class Solution {
    private static String parse(String a){
        String result = "";
        char temp = a.charAt(0);
        int counter = 0;
        int Leng = a.length();
        for(int i = 0;i<Leng;i++){
            char curr = a.charAt(i);
            if(curr == temp){
                counter++;
            }else{
                result+=Integer.toString(counter)+temp;
                temp = a.charAt(i);
                counter = 1;
            }
        }
        result+=Integer.toString(counter)+temp;
        return result;
    }

    public String countAndSay(int n) {
        if(n==1){
            return "1";
        }
        return parse(countAndSay(n-1));
    }
}

File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def parser(self, input: int) -> str:
        input = str(input)
        prev = None
        c = 0
        message = ""
        for curr in input:
            if prev == None:
                prev = curr
                c += 1
            else:
                if curr == prev:
                    c += 1
                else:
                    message += str(c) + prev
                    prev = curr
                    c = 1
        return message + str(c) + prev


    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        return self.parser(self.countAndSay(n-1))

    